docs/src/reviewing_guide.md

```diff
@@ -0,0 +1,95 @@
+   1. E.g. `let length = read_u64(stream); let mut buf = vec![0u8; length];` should be replaced with `let mut buf = vec![0u8; min(length, MAX_LENGTH)];`
+
+
+### Comparing to Diagrams
+When reviewing a pull request, it is important to understand the context of the change. Arguably the easiest way to
+achieve this is to review it in the context of the [diagrams](todo). 
+
+Some questions to ask:
+
+1. Firstly, is the diagram up to date with the code?
+2. Does the PR require the diagram to be updated?
+
+### Catching Common Mistakes
+1. Using `unwrap` or `expect` in production code. These should only be used in tests and in code that is not reachable in production.
+2. As stated before, watch out for parsing of untrusted data, especially when using a third party crate.
+3. When parsing untrusted data it is extremely important that the code is fuzzed and that all branches of the code are covered by tests. Coverage reports are generated whenever code is commited to the development branch, but
+PRs will need to be checked manually if they change this code.
+4. What happens when functions are called with no data, too much data, repeated data, etc.?
+5. Can some flow of data lead to a panic?
+6. When using mutexes, readwrite locks or semaphores, can the lock be held for a long time?
+7. When requesting data from another node, there are a few things to look out for:
+   1. What happens if the node does not respond? Is there a timeout? The timeout should be set as low as possible, otherwise 
+the node can hold up the processing on the requesting node.
+   2. What happens if the node returns unexpected data?
+8. In cases where we have received data from a node, whether we requested it or not, if it is bad, is the node banned?
+
+
+### More specific things to look out for
+
+#### Usize
+`usize` is a platform dependent type. It is 32 bits on 32 bit platforms and 64 bits on 64 bit platforms. This can cause 
+differences in hashing and serialisation. It should be avoided.
+
+#### Atomics
+It is fine to use `AtomicBool`, `AtomicUsize` and the other atomic types, but all `Orderings` must use `Ordering::SeqCst`.
+See [nomicon](https://doc.rust-lang.org/nomicon/atomics.html) for more detail, but most of the ordering enum values have little visible effect on intel based architectures 
+and are only seen in arm based systems. It’s also unlikely that any performance is gained from using a different ordering, so rather be on the safe side and use `SeqCst` everywhere.
+
+It is also pretty much impossible to test ordering on intel based systems, so it is best to avoid it.
+
+#### Vec
+When using `Vec::with_capacity`, is the size inputted provided by an untrusted party? If so, use a maximum value.
+
+#### Unchecked arithmetic and overflows
+Any plus/minus `+/-` has a chance of overflowing or underflowing. It is best to use the checked versions `checked_add` and `checked_sub` to avoid this.
```

Comment:
```suggestion
Any arithmetic operation has a chance of overflowing or underflowing. It is best to use the checked versions, for example `checked_mul`, `checked_sub` etc. to avoid this.
```